_posts/2023-09-19-virtual-thread-1.adoc

```diff
@@ -0,0 +1,281 @@
+---
+layout: post
+title: 'When Quarkus meets Virtual Threads'
+date: 2023-09-19
+tags: virtual-threads
+synopsis: 'What do virtual threads change when building concurrent applications with Quarkus.'
+author: cescoffier
+---
+
+Java 21 offers a new feature that will reshape the development of concurrent applications in Java. 
+For over two years, the Quarkus team explored integrating this new feature to ease the development of distributed applications, including microservices and event-driven applications.
+
+This blog post is the first part of a series of posts and videos demonstrating how to use virtual threads in Quarkus applications. 
+The series covers REST, messaging, containers, native compilation, and our plans for the future.
+But first, let's look at virtual threads, what they change, and what you should know about them.
+
+## The Java world before Java 21
+
+At the beginning of the Java time, Java had _green threads_. 
+Green threads were user-level threads scheduled by the Java virtual machine (JVM) instead of natively by the underlying operating system (OS).
+They emulated multithreaded environments without relying on native OS abilities. 
+They were managed in user space instead of kernel space, enabling them to work in environments that do not have native thread support.
+Green threads were briefly available in Java between 1997 and 2000.
+I used green threads; they did not leave me with a fantastic memory.
+
+In Java 1.3, released in 2000, Java made a big step forward and started integrating OS threads.
+So, the threads are managed by the operating system. 
+It is still the model we are using today.
+Each time a Java application creates a thread, a platform thread is created, which wraps an OS thread.
+So, creating a platform thread creates an OS thread, and **blocking a platform thread blocks an OS thread**. 
+
+When you use a Java framework, you rarely create threads yourself. 
+It is done for you by the framework. 
+For example, when your application receives an HTTP request, the framework creates or reuses a platform thread (and so an OS thread) and executes the processing on that thread.
+The whole processing runs on this thread, and the thread cannot be reused until the processing completes (so the response is sent back). 
+When the processing executes a blocking I/O operation, like calling another service, writing to the file system, or interacting with a database, the thread is blocked, waiting for the response. 
```

Comment:
Do you think a sequence diagram or other visual aid would help here?

Reply:
I'm starting to have some ideas...

Reply:
+1 on visuals to show the various concurrency models in play here

Reply:
Just added pictures.

Reply:
I've added more pictures.